### PR TITLE
Extend sink jms connector to support converter classes from runtime and add protobuf sink converter

### PR DESCRIPTION
**Pull request by meghanayofficial**
This PR is responsible for extending below features:

1. The extension of the converters from the configuration properties and also via kcql using WITHFORMAT and WITHCONVERTER

2. Add Protobuf Messaging converter for sink connector. 

3. Passing Headers via record

4. Supporting subscription name for the source connector

### CODE REVIEW

File: kafka-connect-jms/src/main/scala/com/datamountaineer/streamreactor/connect/jms/sink/converters/ProtoStoredAsConverter.scala [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+import java.util.Collections
+import java.util.concurrent.ConcurrentHashMap
+import java.util.stream.Collectors
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+case class ProtoStoredAsConverter() extends ProtoConverter with StrictLogging {
+
+  private val SCHEMA_PROTO_PATH = "proto_path"
+  private val SCHEMA_PROTO_FILE = "proto_file"
```
Comment (michaelpearce-gain) [inline kafka-connect-jms/src/main/scala/com/datamountaineer/streamreactor/connect/jms/sink/converters/ProtoStoredAsConverter.scala:27]: should probably be dot separated as they are settings used in connector config  for the val, e.g. proto.path and proto.file

Comment (meghanayofficial) [reply kafka-connect-jms/src/main/scala/com/datamountaineer/streamreactor/connect/jms/sink/converters/ProtoStoredAsConverter.scala:27]: using dot separation in STOREDAS has some lexical parser issues so used underscrore instead!

**Review by davidsloan — APPROVED** [review]
Thank you for the contribution.

**Comment by davidsloan** [comment]
Replaced by #855 which has now been merged into master.

**Pull request closed without merging**